File: API/Features/Embarkation/ViewModels/EmbarkationVM.cs
using System.Collections.Generic;

namespace API.Features.Embarkation {

    public class EmbarkationVM {

        public string RefNo { get; set; }
        public string TicketNo { get; set; }
        public string Remarks { get; set; }
        public string Customer { get; set; }
        public string Destination { get; set; }
        public string Driver { get; set; }
        public string Ship { get; set; }
        public int TotalPersons { get; set; }
        public int EmbarkedPassengers { get; set; }
        public string EmbarkationStatus { get; set; }

        public int[] PassengerIds { get; set; }

        public List<EmbarkationPassengerVM> Passengers { get; set; }

    }

}

File: FrontEnd/src/app/features/embarkation/classes/view-models/list/embarkation-vm.ts
import { SimpleEntity } from 'src/app/shared/classes/simple-entity'
import { EmbarkationPassengerVM } from './embarkation-passenger-vm'

export interface EmbarkationVM {

    refNo: string
    ticketNo: string
    customer: SimpleEntity
    destination: SimpleEntity
    pickupPoint: SimpleEntity
    driver: SimpleEntity
    port: SimpleEntity
    ship: SimpleEntity
    totalPersons: number
    embarkedPassengers: number
    embarkationStatus: boolean
    isCheckedIn: string
    remarks: string
    passengerIds: number[]

    passengers: EmbarkationPassengerVM[]

}
